src/transaction.ts

```diff
@@ -161,7 +197,53 @@ class Transaction extends DatastoreRequest {
         : () => {};
     const gaxOptions =
       typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
-    this.#runCommit(gaxOptions, callback);
+    const resolver: Resolver<google.datastore.v1.ICommitResponse> = resolve => {
+      this.#runCommit(
+        gaxOptions,
+        (err?: Error | null, resp?: google.datastore.v1.ICommitResponse) => {
+          resolve({err, resp});
+        }
+      );
+    };
+    this.#withBeginTransaction(gaxOptions, resolver).then(
+      (response: UserCallbackData<google.datastore.v1.ICommitResponse>) => {
+        callback(response.err, response.resp);
+      }
+    );
+  }
+
+  /**
+   * If the transaction has not begun yet then this function ensures the transaction
+   * has started before running the resolver provided. The resolver is a function with one
+   * argument. This argument is a function that is used to pass errors and
+   * response data back to the caller of the withBeginTransaction function.
+   *
+   * @param {CallOptions | undefined} [gaxOptions]
+   * @param {Resolver<T>} [resolver]
+   * @returns {Promise<UserCallbackData<T>>} Returns a promise that will run
```

Comment:
In `#withBeginTransaction`, `{err}` is `UserCallbackData<T>` type and `await new Promise(resolver)` is also of type `UserCallbackData<T>`. Note that `return await new Promise(resolver)` doesn't actually return a promise because the `await` gets the value that the promise resolves to and then the return returns that value. Since the function is `async`, the value provided after `return` is automatically wrapped in a promise and the promise is returned to caller of this function.

Note that when this function runs the transaction and then runs the resolver code, like a transaction get or commit for example, then only the results from the resolver code will be returned to the user. If we delivered the `RunResponse` data back to the user then the signatures of get, runQuery, runAggregationQuery and commit would have to change. The user doesn't have to know about the `RunResponse` data anyway. If the transaction run call results in an error however then the error will be delivered to the user.